Emby.Server.Implementations/Library/DefaultAuthenticationProvider.cs

```diff
@@ -86,19 +173,39 @@ public string GetEmptyHashedString(User user)
             return GetHashedString(user, string.Empty);
         }
 
+        public string GetHashedStringChangeAuth(string NewPassword, PasswordHash passwordHash)
+        {
+            return BitConverter.ToString(_cryptographyProvider.ComputeHash(passwordHash.Id, Encoding.UTF8.GetBytes(NewPassword), passwordHash.SaltBytes)).Replace("-", string.Empty);
+        }
+
         /// <summary>
         /// Gets the hashed string.
         /// </summary>
-        public string GetHashedString(User user, string str)
-        {
-            var salt = user.Salt;
-            if (salt != null)
+        public string GetHashedString(User user, string str)
+        {
+            //This is legacy. Deprecated in the auth method.
+            //return BitConverter.ToString(_cryptoProvider2.ComputeSHA1(Encoding.UTF8.GetBytes(str))).Replace("-", string.Empty);
+            PasswordHash passwordHash;
+            if (String.IsNullOrEmpty(user.Password))
+            {
+                passwordHash = new PasswordHash(_cryptographyProvider);
+            }
+            else
             {
-                // return BCrypt.HashPassword(str, salt);
+                ConvertPasswordFormat(user);
+                passwordHash = new PasswordHash(user.Password);
+            }
+            if (passwordHash.SaltBytes != null)
+            {
+                return BitConverter.ToString(_cryptographyProvider.ComputeHash(passwordHash.Id, Encoding.UTF8.GetBytes(str), passwordHash.SaltBytes)).Replace("-",string.Empty);

```

Comment:
I see this line a lot here with at most a few varying things, can this be extracted in a common method so we don't have so long lines anymore?

Reply:
Yes. Fixed